language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/dependencygraph/TransitiveClosure.scala

```diff
@@ -14,6 +14,13 @@ import com.daml.lf.iface.InterfaceType
   * Ledger API.
   */
 final case class TransitiveClosure(
-    interfaceTypes: Vector[(Identifier, InterfaceType)],
+    orderedDependencies: Vector[(Identifier, Either[DefInterface.FWT, InterfaceType])],
     errors: List[UnsupportedTypeError],
-)
+) {
+
+  lazy val (interfaces, serializableTypes) = orderedDependencies.partitionMap {
+    case (id, Left(interface)) => Left(id -> interface)
+    case (id, Right(serializable)) => Right(id -> serializable)
+  }
```

Comment:
This should probably be split upstream, there's little sense in keeping these two together, I think.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/13557/commits/9414b9e282b38b7c856ed4dc7ccdfeb7a7425f2f